### PR TITLE
Key value store that keeps recently accessed values in a in-memory lru cache for faster access

### PR DESCRIPTION
**Pull request by wmathurin**
* Added class MemCacheKeyValueStore (wraps around regular key value store)
* Added interface KeyValueStore (implemented by both KeyValueEncryptedFileStore and MemCacheKeyValueStore)
* Added tests
* Move smartstore tests from package androidsdk.store to androidsdk.smartstore.store to allow access to default members

### CODE REVIEW

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/MemCachedKeyValueStore.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    @Override
+    public InputStream getStream(String key) {
+        String value = getValue(key);
+        return value == null ?  null : new ByteArrayInputStream(value.getBytes(StandardCharsets.UTF_8));
+    }
+
+    @Override
+    public boolean saveValue(String key, String value) {
+        if (keyValueStore.saveValue(key, value)) {
```
Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/MemCachedKeyValueStore.java:71]: Could be simplified to something like the following, maybe:
```
return (keyValueStore.saveValue(key, value) ? memCache.put(key, value) : false);
```

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/MemCachedKeyValueStore.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    @Override
+    public boolean saveStream(String key, InputStream stream) throws IOException {
+        String value = Encryptor.getStringFromStream(stream);
+        return saveValue(key, value);
+    }
+
+    @Override
+    public boolean deleteValue(String key) {
+        if (keyValueStore.deleteValue(key)) {
```
Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/MemCachedKeyValueStore.java:87]: Maybe similar simplification like above.

**Review by bhariharan — APPROVED** [review]

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/MemCachedKeyValueStore.java (hunk 1 above)
Comment (brandonpage) [reply libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/MemCachedKeyValueStore.java:71]: I think it's fine as is.  `memCache.put(key, value)` returns the previously stored value, not boolean.

**Review by brandonpage — APPROVED** [review]
👍  LGTM.  Maybe before 9.1 goes out I can add a button to the KV Store Inspector to print LRU Cache snapshot on MemeCache Stores.

**Pull request merged**